sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -279,6 +280,7 @@ abstract class Optimizer(catalogManager: CatalogManager)
       RewritePredicateSubquery.ruleName ::
       NormalizeFloatingNumbers.ruleName ::
       ReplaceUpdateFieldsExpression.ruleName ::
+      RewriteCollationJoin.ruleName ::
```

Comment:
@cloud-fan should `RewriteCollationJoin` be a nonExcludable rule or not - not sure what's the convention here? since there is essentially no impact on correctness, I think we may remove it from `nonExcludableRules`